crates/hc_sandbox/src/cmds.rs

```diff
@@ -117,33 +117,38 @@ pub struct Existing {
 }
 
 impl Existing {
-    pub fn load(mut self) -> anyhow::Result<Vec<PathBuf>> {
-        let sandboxes = crate::save::load(std::env::current_dir()?)?;
+    /// Determine all sandbox paths to use based on .hc and given options.
+    /// Returns at minimum all paths in self.existing_paths.
+    pub fn load(&self) -> anyhow::Result<Vec<PathBuf>> {
+        let saved_sandboxes = crate::save::load(std::env::current_dir()?)?;
+        let mut provided = self.existing_paths.clone();
         if self.all {
-            // Get all the sandboxes
-            self.existing_paths.extend(sandboxes)
-        } else if self.last && sandboxes.last().is_some() {
-            // Get just the last sandbox
-            self.existing_paths
-                .push(sandboxes.last().cloned().expect("Safe due to check above"));
+            // Add all sandboxes in .hc
+            provided.extend(saved_sandboxes);
+        } else if self.last {
+            // Get the last sandbox in .hc
+            if let Some(last) = saved_sandboxes.last() {
+                provided.push(last.clone());
+            }
         } else if !self.indices.is_empty() {
             // Get the indices
-            let e = self
-                .indices
-                .into_iter()
-                .filter_map(|i| sandboxes.get(i).cloned());
-            self.existing_paths.extend(e);
-        } else if !self.existing_paths.is_empty() {
-            // If there is existing paths then use those
-        } else if sandboxes.len() == 1 {
-            // If there is only one sandbox then use that
-            self.existing_paths
-                .push(sandboxes.last().cloned().expect("Safe due to check above"));
-        } else if sandboxes.len() > 1 {
-            // There is multiple sandboxes, the use must disambiguate
+            for i in self.indices.clone() {
+                let Some(selected) = saved_sandboxes.get(i) else {
+                    msg!("Warning. No sandbox found at index {}.", i);
+                    continue;
+                };
+                provided.push(selected.clone());
+            }
+        } else if !provided.is_empty() {
+            // If there are existing paths provided then use those.
```

Comment:
What if these paths are gone? 

It seems like what's actually happening is way worse than I thought. When i do something like remove an existing sandbox directory:

```shell
rm -rf /tmp/nix-shell.hB9ALf/N0h2wltj-nlw1vAeABs1y
```

and then try to run it as an existing conductor:

```shell
hc s run -e /tmp/nix-shell.hB9ALf/N0h2wltj-nlw1vAeABs1y
```

Then the sandbox is actually recreated at that existing path.

If I originally created the sandbox with `hc s generate` then I would have had an app installed but now with the newly created conductor, it will have no apps installed. That is a recipe for a confused and rightly frustrated user.

If I specify an existing path directly and it doesn't exist then it shouldn't try to use it.